docs/changelog-fragments/636.feature.rst

```diff
@@ -0,0 +1 @@
+Bumped the libssh version to latest release 0.11.1 -- by :user:`Jakuje`.
```

Comment:
Perhaps, mention that it's for manylinux? macOS is pulling the latest version available in Homebrew, currently: https://github.com/ansible/pylibssh/actions/runs/10801441579/job/29961681318#step:5:212. I've been meaning to figure out compiling it in a more controllable way, similar to manylinux but never got to it...

Reply:
Thanks! Updated.